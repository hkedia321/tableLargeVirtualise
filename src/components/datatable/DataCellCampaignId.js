import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleRight } from '@fortawesome/free-solid-svg-icons'

const DataCellStyled = styled.div`
    position: relative;
    padding: 20px;
`;

const FontAwesomeIconStyled = styled(FontAwesomeIcon)`
    padding-right: 10px;
    width: 20px;

`;

const TextIdStyled = styled.span`
    width: calc(100% - 25px);
    display: inline-block;

`;

const DataCellCampaignId = React.memo(function(props) {
    const { text, isOpen } = props;
    return <DataCellStyled className="dataCellCampaignId" style={props.style} data-testid="dataCellCampaignId">
        {isOpen && <>
        <FontAwesomeIconStyled size={'lg'} color='#3F51B5' icon={faAngleRight} />
            <TextIdStyled>{text}</TextIdStyled>
            </>}
        </DataCellStyled>
})

export default DataCellCampaignId;

DataCellCampaignId.propTypes = {
    text: PropTypes.string,
    isOpen: PropTypes.bool
}