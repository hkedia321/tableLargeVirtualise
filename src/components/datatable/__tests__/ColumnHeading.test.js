import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import DataCellCampaignId from '../DataCellCampaignId';

afterEach(cleanup);

const dataCellCampaignIdProps = {
    text: 'campaign id text', 
    isOpen:true
}

it('renders dataCellCampaignId correctly', () => {
    
    const tableRow = document.createElement('tr');
    const { getByTestId, debug } = render(<DataCellCampaignId {...dataCellCampaignIdProps} />, {
        container: document.body.appendChild(tableRow),
    });
    // debug();
    expect(getByTestId('dataCellCampaignId'));
    // should have text on open state
    expect(getByTestId('dataCellCampaignId')).toHaveTextContent(dataCellCampaignIdProps.text);
})

it('renders dataCellCampaignId correctly', () => {
    
    const tableRow = document.createElement('tr');
    const { getByTestId } = render(<DataCellCampaignId {...dataCellCampaignIdProps} isOpen = {false} />, {
        container: document.body.appendChild(tableRow),
    });
    expect(getByTestId('dataCellCampaignId'));
    // shouln't have text on close state
    expect(getByTestId('dataCellCampaignId')).not.toHaveTextContent(dataCellCampaignIdProps.text);
})

it('matches snapshot', () => {
    const tree = renderer.create(<DataCellCampaignId {...dataCellCampaignIdProps} />).toJSON();
    expect(tree).toMatchSnapshot();
})