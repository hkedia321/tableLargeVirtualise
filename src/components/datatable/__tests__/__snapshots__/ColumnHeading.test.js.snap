// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c0 {
  position: relative;
  padding: 20px;
}

.c1 {
  padding-right: 10px;
  width: 20px;
}

.c2 {
  width: calc(100% - 25px);
  display: inline-block;
}

<td
  className="c0"
  data-testid="dataCellCampaignId"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right fa-w-8 fa-lg c1"
    color="#3F51B5"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 256 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="c2"
  >
    campaign id text
  </span>
</td>
`;
